pandas/tests/test_strings.py

```diff
@@ -435,6 +435,12 @@ def test_replace(self):
                 for data in (['a', 'b', None], ['a', 'b', 'c', 'ad']):
                     values = klass(data)
                     self.assertRaises(TypeError, values.str.replace, 'a', repl)
+        
+        # GH 15055, callable repl
+        repl = lambda m: m.group(0).swapcase()
+        result = values.str.replace('[a-z][A-Z]{2}', repl, n=2)
+        exp = Series([u('foObaD__baRbaD'), NA])
```

Comment:
test where the callable has different than 1 arg (should be error)

test using named field in re (and callable works correctly)